db-engine/src/main/java/com/splicemachine/db/impl/sql/catalog/DataDictionaryImpl.java

```diff
@@ -1152,11 +1152,16 @@ public void addDescriptor(TupleDescriptor td,
                               TupleDescriptor parent,
                               int catalogNumber,
                               boolean duplicatesAllowed,
-                              TransactionController tc) throws StandardException{
+                              TransactionController tc,
+                              boolean buildVersion2Descriptor) throws StandardException{
         TabInfoImpl ti=(catalogNumber<NUM_CORE)?coreInfo[catalogNumber]:getNonCoreTI(catalogNumber);
 
         ExecRow row=ti.getCatalogRowFactory().makeRow(td,parent);
 
+        if (buildVersion2Descriptor)
```

Comment:
Do you mean DataDictionary.SYSTABLES_CATALOG_NUM?  I only have one location that passes buildVersion2Descriptor as true.  In CreateTableConstantOperation.createTable.  What is the concern here?

Reply:
Yes, I meant DataDictionary.SYSTABLES_CATALOG_NUM. This is just a safety check to prevent mis-usage. Since this flag only affects systable rows, and the combination of "other table type + true" will cause a bad row to be rewritten, I suggest to add this check. It is not a big deal though.

Reply:
OK @yxia92 , I made the change.